daylay92/stacktrace-backend
# Add basic configuration to the App

#### What does this PR do?
Add most of the essential configurations for the Application.
#### Description of Task to be completed?
- Setup continuous integration with travis ci and test coverage analysis with coveralls.
- Add code linting with Airbnb style guide and ESlint.
- Setup test script 
- Setup mongodb database for testing, development and production environments.
- Add hound CI for code quality review of PR.
- Write unit test.
#### How should this be manually tested?
- Clone the repo and cd into the project folder
- Checkout to `ch-basic-config` branch and run `npm install`.
- Add a `.env` file to the root directory and include in it the following environment variables;
```
 MONGO_DEV_URI=your mongodb connection string for development
MONGO_URI=your mongodb connection string for the test environment
```
- Run `npm test` to see results. Also, inspect the file changes.
